### PR TITLE
Added documentation for goal preemption on behavior trees

### PR DESCRIPTION
**Pull request by gimait**
Signed-off-by: Aitor Miguel Blanco <aitormibl@gmail.com>

* Added new condition node to plugins index (navigation2 #[1712](https://github.com/ros-planning/navigation2/pull/1712)).
* Updated migration guide.
* Added new section for recommended tree structures in concepts, under Behavior Trees.

Other changes:
* Updated links in plugins index.
* Small typos and blank spaces at end of lines.

### CODE REVIEW

File: concepts/index.rst [hunk 1]
```diff
@@ -73,17 +73,44 @@ Additionally, modeling choices like to shoot at the goal from the left, right, o
...
+Cancel Action on Condition
+--------------------------
+
+The default behavior trees on the stack use the following tree structure to stop recovery behaviors if the navigation goal is preempted:
+
+.. image:: images/behavior_tree/recovery_subtree.png
+  :width: 50%
+  :align: center
+
+This tree structure allows having an asychronous node running (in this case the RecoveryActions node) while checking a condition (i.e., the navigation goal has been updated).
```
Comment (SteveMacenski) [inline concepts/index.rst:108]:
```suggestion
This tree structure allows having an asychronous node running (in this case the RecoveryActions node) while checking a condition (i.e., the navigation goal has been updated).
This allows for preemption of long-running tasks when a new goal comes in.
```

File: concepts/index.rst [hunk 2]
```diff
@@ -73,17 +73,44 @@ Additionally, modeling choices like to shoot at the goal from the left, right, o
...
 The node plugins are loaded into the BT and when the XML file of the tree is parsed, the registered names are associated.
 At this point, we can march through the behavior tree to navigate.
 
 One reason this library is used is its ability to load subtrees. This means that the Navigation2 behavior tree can be loaded into another higher-level BT to use this project as node plugin.
 An example would be in soccer play, using the Navigation2 behavior tree as the "go to ball" node with a ball detection as part of a larger task.
 Additionally, we supply a ``NavigateToPoseAction`` plugin for BT so the Navigation2 stack can be called from a client application through the usual action interface.
 
+.. _useful-tree-structures:
+
+Useful Tree Structures
```
Comment (SteveMacenski) [inline concepts/index.rst:89]: I don't think this is the place for this section. There's another ticket for adding documentation about BTs/designers guides/etc. This is just a quick introduction to the nomenclature and concepts.

Please remove this section. Maybe add this text to that ticket to add to those docs (or start on those separate docs, if you're interested :smile: ). This PR is just about the migration guide. Without this section, we can merge this PR

Comment (gimait) [reply concepts/index.rst:89]: I'll revert all changes in this file and take a look at that ticket. If nobody is working on it maybe I'll start with it.

File: migration/Eloquent.rst [hunk 3]
```diff
@@ -33,14 +33,16 @@ DWB Contains new parameters as an update relative to the ROS1 updates, `see here
...
 
 Many new behavior tree nodes were added.
 These behavior tree nodes are now BT plugins and dynamically loadable at run-time using behavior tree cpp v3.
-See ``nav2_behavior_tree`` for a full listing, or :ref:`plugins` for the current list of behavior tree plugins and their descriptions. 
+The default behavior trees have been upgraded to stop the recovery behaviours and trigger a replanning when the navigation goal is preempted.
+See :ref:`useful-tree-structures` for detailed explanation of the behaviour tree structures used in this upgrade.
+See ``nav2_behavior_tree`` for a full listing, or :ref:`plugins` for the current list of behavior tree plugins and their descriptions.
 These plugins are set as default in the ``nav2_bt_navigator`` but may be overridden by the ``bt_plugins`` parameter to include your specific plugins.
 
 Original GitHub tickets:
```
[the remark this answers is not in the document]
Comment (gimait) [reply migration/Eloquent.rst:48]: yes, of course

File: concepts/index.rst (hunk 2 above)
Comment (SteveMacenski) [reply concepts/index.rst:89]: That would be great! I've asked someone else to also help out, but there's plenty to go around (and he hasn't agreed yet). I look at this like an entire subsection of the website at a high level like "tutorials" that contains several pages.

**Comment by gimait** [comment]
I reverted the changes in concepts index (except from the typo on BT Navigator, and the whitespaces) and added the goalupdated condition pr to the migration guide.

**Pull request merged**